src/status.rs

```diff
@@ -596,6 +599,22 @@ impl From<StatusCode> for u16 {
     }
 }
 
+#[cfg(feature = "compat")]
+impl From<http_types::StatusCode> for StatusCode {
+    fn from(status: http_types::StatusCode) -> StatusCode {
+        StatusCode::try_from(status.as_u16())
```

Comment:
Thanks to `hyper::StatusCode::Unregistered` being able to have any stupid `u16`, this is likely the best approach!